### PR TITLE
fix: stop manually starting a companion process

### PR DESCRIPTION
**Pull request by rerorero**
This PR proposes to stop manually starting the companion process. Instead, you can let idb command manage the connections.

`idb` command looks for a running companion process associated with the given udid and it spanws a new companion process if not found. Also it just uses the existing one if found. Once it connects to the companion, idb stores the association info in its state (/tmp/idb/state). This state is not created when you manually run `idb_companion`. 
`idb` lets companion  to use unix domain socket instead of tcp if port is not specified. Because idb creates a unique UDS file for each udid, you don't need to care about the port conflicts e.g. `/tmp/idb/42372A50-5AF6-472F-973A-3F4C6DE56255_companion.sock`

I'd expect that this fixes the [address already in use error](https://github.com/appium/appium-idb/pull/63#discussion_r1049792518) ~~and also fixes this [CI issue](https://github.com/appium/appium-idb/actions/runs/3705525505/jobs/6299284956) since it tried to connect via UDS somehow while a companion process listens on tcp.~~ > this is another issue, see https://github.com/appium/appium-idb/pull/66

However, [as the document mentions](https://fbidb.io/docs/commands/#starting-a-grpc-server), using grpc enables you to expose a companion over the network to another machine. If this is needed, it'd be better to make grpc port configurable instead of this approach.

### CODE REVIEW

**Comment by rerorero** [comment]
The connection issue has not been fixed. I'll take a look deeper into it.
```
Error: The device '5C59AF64-552A-464A-B042-6A162876029E' is not responding to idb requests after 60000ms timeout. Original error: Failed to connect to companion at address DomainSocketAddress(path='/tmp/idb/5C59AF64-552A-464A-B042-6A162876029E_companion.sock')
```

**Review by KazuCocoa — APPROVED** [review]
sounds reasonable to me

**Comment by rerorero** [comment]
Probably test will fixed by merging this https://github.com/appium/appium-idb/pull/63

File: test/functional/idb-e2e-specs.js [hunk 1]
```diff
@@ -75,8 +73,8 @@ describe('idb general', function () {
       } catch (e) {}
     });
 
-    it('should not be able to call connect multiple times', async function () {
-      await idb.connect().should.be.rejected;
+    it('should be able to call connect multiple times', async function () {
+      await idb.connect().should.be.fulfilled;
```
Comment (rerorero) [inline test/functional/idb-e2e-specs.js:77]: The second connect() call just uses the same UDS.

**Comment by rerorero** [comment]
hmm.. xctest tests seems flakey. Let me extend the retry interval.

**Comment by rerorero** [comment]
Now the test has passed, please take a look.

**Review by KazuCocoa — APPROVED** [review]
thank you

**Pull request merged**